File: config/config-sbc2ha/src/main/java/com/dfi/sbc2ha/config/sbc2ha/definition/sensor/SingleSourceConfig.java
package com.dfi.sbc2ha.config.sbc2ha.definition.sensor;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public abstract class SingleSourceConfig extends ScalarSensorConfig {
    /**
     * The sensor to read values.
     * (Required, ID):
     */
    private String sensor;

    public SingleSourceConfig(String sensor) {
        super();
        this.sensor = sensor;
    }

    public SingleSourceConfig() {
        super();
    }

    public String getSensor() {
        return sensor;
    }
}

File: config/editor/src/app/definition/sensor/SingleSourceConfig.ts
import {ScalarSensorConfig} from "./ScalarSensorConfig";
import {PlatformType} from "../enums/PlatformType";


export abstract class SingleSourceConfig extends ScalarSensorConfig {
    sensor: string;

    protected constructor(platform: PlatformType, name: string, sensor: string) {
        super(platform, name);
        this.sensor = sensor;
    }
}
